PWGLF/Tasks/cascadecorrelations.cxx

```diff
@@ -0,0 +1,234 @@
+      if (TMath::Abs(bachTrack.tpcNSigmaPi()) < 3) {
+        if (TMath::Abs(bachTrack.tpcNSigmaKa()) < 3) {
+          // TODO: ambiguous! ignore for now
+          cascflags(0);
+          continue;
+        }
+        cascflags(1);
+        continue;
+      } else if (TMath::Abs(bachTrack.tpcNSigmaKa()) < 3) {
+        cascflags(2);
+        continue;
+      }
+      // if we reach here, the bachelor was neither pion nor kaon
+      cascflags(0);
+    } // cascade loop
+  }   // process
+};    // struct
+
+struct cascadeCorrelations {
+  HistogramRegistry registry{
+    "registry",
+    {
+      {"hMassXiMinus", "hMassXiMinus", {HistType::kTH1F, {{3000, 0.0f, 3.0f, "Inv. Mass (GeV/c^{2})"}}}},
+      {"hMassXiPlus", "hMassXiPlus", {HistType::kTH1F, {{3000, 0.0f, 3.0f, "Inv. Mass (GeV/c^{2})"}}}},
+      {"hMassOmegaMinus", "hMassOmegaMinus", {HistType::kTH1F, {{3000, 0.0f, 3.0f, "Inv. Mass (GeV/c^{2})"}}}},
+      {"hMassOmegaPlus", "hMassOmegaPlus", {HistType::kTH1F, {{3000, 0.0f, 3.0f, "Inv. Mass (GeV/c^{2})"}}}},
+      {"hPhi", "hPhi", {HistType::kTH1F, {{100, 0, 2 * PI, "#varphi"}}}},
+      {"hDeltaPhiSS", "hDeltaPhiSS", {HistType::kTH1F, {{100, -PI / 2, 1.5 * PI, "#Delta#varphi"}}}},
+      {"hDeltaPhiOS", "hDeltaPhiOS", {HistType::kTH1F, {{100, -PI / 2, 1.5 * PI, "#Delta#varphi"}}}},
+    },
+  };
+
+  Filter Selector = aod::cascadeflags::isSelected > 0; // TODO: treat Omega's and Xi's differently
+
+  void process(soa::Join<aod::Collisions, aod::EvSels>::iterator const& collision, soa::Filtered<aod::CascDataExtSelected> const& Cascades, aod::V0sLinked const&, aod::V0Datas const&, FullTracksExtIU const&)
+  {
+    // Some QA on the cascades
+    for (auto& casc : Cascades) {
+
+      auto v0 = casc.v0_as<o2::aod::V0sLinked>();
+      if (!(v0.has_v0Data())) {
+        continue; // reject if no v0data
+      }
+      auto v0data = v0.v0Data();
+
+      if (casc.sign() < 0) { // FIXME: could be done better...
+        registry.fill(HIST("hMassXiMinus"), casc.mXi());
+        registry.fill(HIST("hMassOmegaMinus"), casc.mOmega());
+      } else {
+        registry.fill(HIST("hMassXiPlus"), casc.mXi());
+        registry.fill(HIST("hMassOmegaPlus"), casc.mOmega());
+      }
+
+      registry.fill(HIST("hPhi"), RecoDecay::phi(casc.px(), casc.py()));
+    } // casc loop
+
+    for (auto& [c0, c1] : combinations(Cascades, Cascades)) { // combinations automatically applies strictly upper in case of 2 identical tables
+      auto lambda0 = c0.v0_as<o2::aod::V0sLinked>();
+      auto lambda1 = c1.v0_as<o2::aod::V0sLinked>();
+      if (!(lambda0.has_v0Data()) || !(lambda1.has_v0Data())) {
+        continue; // reject if no v0data in either of the lambda's
+      }
+      auto v0data0 = lambda0.v0Data();
+      auto v0data1 = lambda1.v0Data();
+
+      double phi0 = RecoDecay::phi(c0.px(), c0.py());
+      double phi1 = RecoDecay::phi(c1.px(), c1.py());
+      double dphi = std::fmod(phi0 - phi1 + 2.5 * PI, 2 * PI) - 0.5 * PI;
```

Comment:
Is this analogous to the method ComputeDeltaPhi(Double_t phi1, Double_t phi2) used in https://github.com/AliceO2Group/O2Physics/blob/master/PWGLF/Tasks/hStrangeCorrelation.cxx?

Reply:
Yes, it is in principle the same. If desirable I can change it to their format method, or maybe put the method somewhere such that people can easily call it from their analysis task

Reply:
There is already a configurable [RecoDecay::constrainAngle](https://github.com/AliceO2Group/O2Physics/blob/master/Common/Core/RecoDecay.h#L225) function for this, which can be directly fed with `phi0 - phi1`.
Please let's not make thousands versions of this.

Reply:
Hi Vit, I wasn't aware of this method, I've implemented it now, thanks for pointing it out!